src/librbd/io/ImageRequestWQ.h

```diff
@@ -22,6 +22,31 @@ class AioCompletion;
 template <typename> class ImageRequest;
 class ReadResult;
 
+class ImageQosStatus {
```

Comment:
Nit: can you move into an inner struct of ```ImageRequestWQ``` since it's only used by the work queue -- can also probably shorten the name to just ```QoSThrottle```